frontend/src/metabase-lib/metadata.ts

```diff
@@ -14,3 +14,18 @@ export function metadataProvider(
 export function displayName(query: Query, clause: Clause): string {
   return ML_MetadataCalculation.display_name(query, clause);
 }
+
+export type DisplayInfo = {
+  display_name: string;
+  name?: string;
+  table?: {
+    name: string;
+    display_name: string;
+  };
+};
+
+// x can be any sort of opaque object, e.g. a clause or metadata map. Values returned depend on what you pass in, but it
+// should always have display_name... see :metabase.lib.metadata.calculation/display-info schema
+export function displayInfo(query: Query, x: any): DisplayInfo {
+  return ML.display_info(query, x);
+}
```

Comment:
We should avoid using `any` here because it'd let us pass absolutely whatever to `displayInfo`. Think we should just enumerate suitable types here:

```ts
function displayInfo(query: Query, x: Clause | ColumnMetadata) { ... }
```